crates/ruff_linter/src/rules/ruff/rules/chmod_non_octal.rs

```diff
@@ -0,0 +1,177 @@
+use ruff_diagnostics::{Applicability, Edit, Fix};
+use ruff_macros::{ViolationMetadata, derive_message_formats};
+use ruff_python_ast::{self as ast, Expr, ExprCall};
+use ruff_python_semantic::{SemanticModel, analyze};
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+use crate::{FixAvailability, Violation};
+
+/// ## What it does
+/// Checks for `chmod` calls which use non-octal integer literals.
+///
+/// ## Why is this bad?
+///
+/// Numeric modes are made up of one to four octal digits. Converting a non-octal
+/// integer to octal may not be the mode the author intended.
+///
+/// ## Example
+///
+/// ```python
+/// os.chmod("foo", 644)
+/// ```
+///
+/// Use instead:
+///
+/// ```python
+/// os.chmod("foo", 0o644)
+/// ```
+///
+/// ## Fix safety
+///
+/// This rule's fix is marked as unsafe because it changes runtime behavior.
+///
+/// ## Fix availability
+///
+/// A fix is only available if the existing digits could make up a valid octal literal.
+#[derive(ViolationMetadata)]
+pub(crate) struct ChmodNonOctal {
+    reason: Reason,
+}
+
+#[derive(Debug, Clone, Copy)]
+enum Reason {
+    Decimal { found: u16, suggested: u16 },
+    Invalid,
+}
+
+impl Violation for ChmodNonOctal {
+    const FIX_AVAILABILITY: FixAvailability = FixAvailability::Sometimes;
+
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        match self.reason {
+            Reason::Decimal { found, suggested } => {
+                format!("Non-octal mode `{found}`, did you mean `{suggested:#o}`?")
```

Comment:
See my comment below. I don't think we should offer a suggestion here because the promise of the rule is that we believe that the used decimal permissions are actually wrong and simply rewriting to octal doesn't fix that.